Assets/Scripts/Actor/EnemyAttackManager.cs

```diff
@@ -0,0 +1,146 @@
+﻿using JetBrains.Annotations;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Actor
+{
+    public class EnemyAttackManager: MonoBehaviour
+    {
+        /// <summary>
+        /// A list of attack patterns that can contain non-spells and spells.
+        /// </summary>
+        public GameObject[] attacks;
+        
+        /// <summary>
+        /// The attack pattern currently active.
+        /// </summary>
+        public EnemyAttack currentAttack;
+        
+        /// <summary>
+        /// How long to wait before switching to a new attack pattern.
+        /// </summary>
+        public float attackSwitchCooldownSeconds = 2f;
+        
+        /// <summary>
+        /// The score text box to update and display.
+        /// </summary>
+        [CanBeNull] public Text attackNameText;
```

Comment:
This is nullable to fix an edge-case bug where I think we try to set the text box right as we transition the scene, so it ends up being null.